torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -745,6 +745,22 @@ RegisterOperators reg(
            handler(ss.str());
          },
          aliasAnalysisSpecialCase()),
+     OperatorGenerator(
+         TORCH_SELECTIVE_SCHEMA("aten::eq.optional(t1? self, t2 obj) -> bool"),
+         [](Stack* stack) {
+           IValue self, obj;
+           pop(stack, self, obj);
+           push(stack, self.is(obj));
```

Comment:
`is` and `==` have different semantics. I think we might also consider if we want `== None` to refine the type of the inputs if used like `is None` when adding support for this.